src/inspect_viz/view/beta/_scores_radar.py

```diff
@@ -0,0 +1,297 @@
+
+    if not all_rows:
+        raise ValueError("No valid model data found after processing.")
+
+    return pd.concat(all_rows, ignore_index=True)
+
+
+def compute_angles(num_axes: int, endpoint: bool = True) -> NDArray[np.floating[Any]]:
+    """Computes the angles by number of axes."""
+    return np.linspace(0, 2 * np.pi, num_axes, endpoint=endpoint)
+
+
+def labels_coordinates(metrics: list[str]) -> dict[str, list[str] | list[float]]:
+    """Computes coordinates for labels to be used in a radar chart."""
+    angles = compute_angles(len(metrics), endpoint=False)
+    return {
+        "metric": metrics,
+        "x": (1.12 * np.cos(angles)).tolist(),
+        "y": (1.12 * np.sin(angles)).tolist(),
+    }
+
+
+def axes_coordinates(num_axes: int) -> dict[str, list[float]]:
+    """Computes coordinates for axes to be used in a radar chart."""
+    angles = compute_angles(num_axes, endpoint=False)
+    return {
+        "x": (np.tile([0, 1], num_axes) * np.repeat(np.cos(angles), 2)).tolist(),
+        "y": (np.tile([0, 1], num_axes) * np.repeat(np.sin(angles), 2)).tolist(),
+    }
+
+
+def grid_circles_coordinates() -> list[dict[str, list[float]]]:
+    """Computes coordinates for grid circles to be used in a radar chart."""
+    radii = [0.2, 0.4, 0.6, 0.8, 1.0]
+    circle_angles = compute_angles(100)
+    return [
+        {
+            "x": (radius * np.cos(circle_angles)).tolist(),
+            "y": (radius * np.sin(circle_angles)).tolist(),
+        }
+        for radius in radii
+    ]
+
+
+def scores_radar(
+    data: Data | pd.DataFrame,
+    model: str = "model_display_name",
+    title: str | Title | None = None,
+    height: float | None = 700,
+    width: float | None = 700,
+    legend: Legend | NotGiven | None = NOT_GIVEN,
+    **attributes: Unpack[PlotAttributes],
+) -> Component:
+    """
+    Creates a radar chart showing scores for multiple models across multiple metrics.
+
+    Args:
+        data: A dataframe prepared using the `scores_radar_df` function.
+        model: Name of field holding the model (defaults to "model_display_name").
+        title: Title for plot (`str` or mark created with the `title()` function)
+        height: The outer height of the plot in pixels, including margins. Defaults to 700.
+        width: The outer width of the plot in pixels, including margins. Defaults to 700.
+        legend: Options for the legend. Pass None to disable the legend.
+        **attributes: Additional `PlotAttributes`.
+    """
+    if isinstance(data, pd.DataFrame):
+        data = Data.from_dataframe(data)
+
+    if "model_display_name" not in data.columns:
+        model = "model"
+
+    required_columns = [model, "task_id", "log", "metric", "value", "x", "y"]
+    missing_columns = [col for col in required_columns if col not in data.columns]
+    if missing_columns:
+        raise ValueError(f"Required columns not found in data: {missing_columns}")
+
+    metrics = data.column_unique("metric")
```

Comment:
I know we decided that passing metrics to the df function made the most sense, but now I am wondering whether it makes sense to support passing them as an arg here too (e.g. I could prepare a dataframe with all metrics, but then only choose to show some of the metrics)?

Reply:
In that case, this function would need to take a DataFrame as input instead of Data as we would need to filter the rows which we cannot do with Data. There's a specific schema we expect in scores_radar so I'd prefer users to rely on scores_radar_df to prepare their dataframe.